feat: add @namespace validation to no-invalid-at-rule-placement rule (#183)
eslint/css: docs/rules/no-invalid-at-rule-placement.md

- The `@charset` rule must be placed at the very beginning of a stylesheet, before any other rules, comments, or whitespace.
- The `@import` rule must be placed at the beginning of a stylesheet, before any other at-rules (except `@charset` and `@layer` statements) and style rules.

> - The `@namespace` rule must be placed at the beginning of a stylesheet, before any other at-rules (except `@charset` and `@import`) and style rules.

Reviewer: Just to clarify
```suggestion
- The `@namespace` rule must be placed after`@charset` and `@import` and before any other at-rules and style rules.
```

If these rules are placed incorrectly, browsers will ignore them, resulting in potential encoding issues, missing imported styles, or incorrect namespace application.

## Rule Details

This rule warns when it finds:

1. A `@charset` rule that is not the first rule in the stylesheet
2. An `@import` rule that appears after any other at-rules or style rules (except `@charset` and `@layer` statements)

> 3. A `@namespace` rule that appears after any other at-rules or style rules (except `@charset` and `@import`)

Reviewer: ```suggestion
3. A `@namespace` rule that appears before `@charset` or `@import` or after any other at-rules or style rules
```
